src/core/qgsdatadefinedsizelegend.cpp

```diff
@@ -126,7 +126,8 @@ QgsLegendSymbolList QgsDataDefinedSizeLegend::legendSymbolList() const
   QgsLegendSymbolList lst;
   if ( !mTitleLabel.isEmpty() )
   {
-    QgsLegendSymbolItem title( nullptr, mTitleLabel, QString() );
+    // we're abusing the ruleKey field here so we can later identify the resulting legend nodes as data defined size related
+    QgsLegendSymbolItem title( nullptr, mTitleLabel, QStringLiteral( "data-defined-size" ) );
```

Comment:
instead of the abuse, how about we add a QMap< int, QVariant > userData to QgsLegendSymbolItem, and adapt QgsSymbolLegendNode::data to return those values for unknown roles.

Reply:
We could have a new role in `QgsLayerTreeModelLegendNode::CustomRole` for that.
But, why a `QMap< int, QVariant >`?  What would you store there (in this case and/or in future uses of it)?

Reply:
> We could have a new role in QgsLayerTreeModelLegendNode::CustomRole for that, instead of handling all unknown roles.

That also works. I was just thinking ahead if allowing custom roles in QgsLegendSymbolItem would also be useful in future.

> But, why a QMap< int, QVariant >? What would you store there (in this case and/or in future uses of it)?

The int keys would correspond to a role from QgsLayerTreeModelLegendNode (either public or private). I'd find a QMap approach slightly preferable to having separate members for every property we decide to add in future.